test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetVerifyTests.cs

```diff
@@ -0,0 +1,150 @@
+
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NuGet.Common;
+using NuGet.Test.Utility;
+using Test.Utility.Signing;
+using Xunit;
+
+namespace Dotnet.Integration.Test
+{
+    [Collection("Dotnet Integration Tests")]
+    public class DotnetVerifyTests
+    {
+        private readonly string _noTimestamperWarningCode = NuGetLogCode.NU3027.ToString();
+        private readonly string _primarySignatureInvalidErrorCode = NuGetLogCode.NU3018.ToString();
+        private readonly string _noMatchingCertErrorCode = NuGetLogCode.NU3034.ToString();
+        private readonly string _notSignedErrorCode = NuGetLogCode.NU3004.ToString();
+
+        private MsbuildIntegrationTestFixture _msbuildFixture;
+
+        public DotnetVerifyTests(MsbuildIntegrationTestFixture fixture)
+        {
+            _msbuildFixture = fixture;
+        }
+
+        [CIOnlyFact]
+        public async Task Verify_UnSignedPackage_Fails()
+        {
+            using (var packageDir = TestDirectory.Create())
+            {
+                var packageId = "Unsigned.PackageX";
+                var packageVersion = "1.0.0";
+                var packageFile = await TestPackagesCore.GetRuntimePackageAsync(packageDir, packageId, packageVersion);
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    packageDir,
+                    $"nuget verify {packageFile.FullName}",
+                    ignoreExitCode: true);
+
+                result.Success.Should().BeFalse(because: result.AllOutput);
+                result.Output.Should().Contain(_notSignedErrorCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("--all")]
+        [InlineData("")]
+        public void Verify_AuthorSignedAndTimestampedPackage_Succceeds(string optionAll)
+        {
+            // Arrange
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var packageFile = new FileInfo(Path.Combine(testDirectory, "TestPackage.AuthorSigned.1.0.0.nupkg"));
+                var package = SigningTestUtility.GetResourceBytes(packageFile.Name);
+                File.WriteAllBytes(packageFile.FullName, package);
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    testDirectory,
+                    $"nuget verify {packageFile.FullName} {optionAll}",
+                    ignoreExitCode: true);
+
+                result.Success.Should().BeTrue(because: result.AllOutput);
+                result.Output.Should().NotContain(_noTimestamperWarningCode);
+                result.Output.Should().NotContain(_primarySignatureInvalidErrorCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("--certificate-fingerprint 775AAB607AA76028A7CC7A873A9513FF0C3B40DF09B7B83D21689A3675B34D9A")]
+        [InlineData("--certificate-fingerprint ABC --certificate-fingerprint DEF")]
```

Comment:
It's very unlikely that if passing two fingerprints on the command line works, that passing only one will fail. Therefore, this is basically testing the same thing twice, making CI just that little bit slower.

Reply:
I agree Andy. Changed the test to act only on multiple fingerprints. Will push in my next commit.